src/stories/MultiSelect.stories.tsx

```diff
@@ -56,16 +56,7 @@ const WithParentOptionsVisibilityControlTemplate: Story<MultiSelectProps<number,
   const [selectedOptions, setSelectedOptions] = useState<number[]>([]);
   const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
   const [optionsVisible, setOptionsVisible] = useState(false);
-  /*
-    Admittedly, this is a bit confusing. Because the onBlur event within the MultiSelect bubbles up _before_ the
```

Comment:
@karthikbtf let me know what you think. This is less than ideal but I was unable to figure out another way to achieve the functionality given the way the MUI Popover works. Thankfully the `setTimeout` is something the implemented in the parent, and not within the `MultiSelect`. 

If this isn't acceptable, I'm going to have to say we hold off on this functionality for now because I haven't been able to figure out a better solution and I've already spent a bit too long on this.

Reply:
@nickgraz thanks for revisiting, didn't intend for you to spend more time on this with my previous comment.
Interesting interactions/event problem.
This current approach is fine, the previous approach seemed cleaner to me. 
Current approach seems to keep options open even after selecting an item in the dropdown list.
I agree let's not spend much time on making it work.

Reply:
Ah, good catch. I had to ignore the click capture unless it was on the Popover backdrop. I agree the other solution was cleaner, but unfortunately it had a bug that was a symptom of the two events firing sequentially.

Reply:
Got it, this sounds like a good solution to stick with.

Reply:
Thanks for fixing the click issue 🙌